extensions/fairifier-rdf-extension/dtls/fairifier/FileSystemRdfSkeletonImpl.java

```diff
@@ -22,63 +22,74 @@
 public class FileSystemRdfSkeletonImpl implements RdfSkeletonService {
     private static final Path SAVELOCATION = Paths.get(System.getProperty("user.home") + "/.local/share/openrefine/");
     private static final ObjectMapper mapper = new ObjectMapper();
-
+    private static final RdfSkeletonTransformer rdfSkeletonJsonTransformerImpl = new  RdfSkeletonJsonTransformerImpl(); 
     
     @Override
-    public List<String[]> listModels(final String fileType)  throws IOException {
+    public List<RdfSkeletonTransformer> listModels(final String fileType)  throws IOException {
         return Files.list(SAVELOCATION)
-                .filter(Files::isRegularFile)
+                .filter(Files::isRegularFile)                
                 .map(this::getStringAndFiletype)
-                .filter(val -> val[1] != null)
-                .filter(val -> val[2].equals(fileType) )
+                .filter(val -> val.getSkeletonMetadata() != null )
+                .filter(val -> val.getSkeletonMetadata().getFileType().equals(fileType) )
                 .collect(Collectors.toList());
     }
 
     @Override
-    public List<String[]> listModels()  throws IOException {
+    public List<RdfSkeletonTransformer> listModels()  throws IOException {
         return Files.list(SAVELOCATION)
                 .filter(Files::isRegularFile)
                 .map(this::getStringAndFiletype)
-                .filter(val -> val[0] != null)
+                .filter(val -> val.getSkeletonMetadata() != null )
                 .collect(Collectors.toList());
     }
     
     @Override
-    public void saveModel(String json, String fileType, String title,String projectId)  throws IOException {
-        Path fileLocation = Paths.get(SAVELOCATION.toString() + File.separator + projectId + ".skeleton.json");        
-        Files.write(fileLocation, json.getBytes(StandardCharsets.UTF_8));
-        fileLocation = Paths.get(SAVELOCATION.toString() + File.separator + projectId + ".metadata.skeleton.json");
+    public void saveModel(String json, String fileType, String title,String projectId)  throws IOException {   
+        Path fileLocation = Paths.get(SAVELOCATION.toString() + File.separator + projectId + ".metadata.skeleton.json");        
         try (BufferedWriter writer = Files.newBufferedWriter(fileLocation)) {
             SkeletonMetadata metadata = new SkeletonMetadata();
             metadata.setFileType(fileType);
             metadata.setTitle(title);
+            metadata.setFileName(fileLocation.toString());
+            metadata.setProjectId(projectId);
+            rdfSkeletonJsonTransformerImpl.setModelMetadata(metadata);
+            rdfSkeletonJsonTransformerImpl.transform(json);
+            
             mapper.writeValue(writer, metadata);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
-        }
+        }      
+        json = rdfSkeletonJsonTransformerImpl.getModelAsJsonString();
```

Comment:
The codacy check warns about reassigning the json variable. I'm not sure whether that's an issue in this case, but let's have a discussion about it.